src/wallet/wallet.cpp

```diff
@@ -3184,6 +3184,24 @@ bool CWallet::AttachChain(const std::shared_ptr<CWallet>& walletInstance, interf
 
     if (tip_height && *tip_height != rescan_height)
     {
+        // No need to read and scan block if block was created before
+        // our wallet birthday (as adjusted for block time variability)
+        std::optional<int64_t> time_first_key;
+        for (auto spk_man : walletInstance->GetAllScriptPubKeyMans()) {
+            int64_t time = spk_man->GetTimeFirstKey();
+            if (!time_first_key || time < *time_first_key) time_first_key = time;
+        }
+        if (time_first_key) {
+            FoundBlock found = FoundBlock().height(rescan_height);
```

Comment:
what is the reason behind the height set here? It doesn't seems to be used anywhere inside `findFirstBlockWithTimeAndHeight` nor below.

Reply:
It is used in `chain.getBlockHash(rescan_height)`?

Reply:
yeah ok, digested it now. thanks. Was expecting `FoundBlock` to be an inputs-only container when it's actually a function's outputs container.